streamline TestOthersItems (ERC)

This patch changes a deeply nested "if" into a series of "breaks",
for better readability and to facilitate future extensions.

- Werner

---

Index: kicad/eeschema/erc.cpp
===================================================================
--- kicad.orig/eeschema/erc.cpp	2009-07-03 05:00:34.000000000 -0300
+++ kicad/eeschema/erc.cpp	2009-07-03 05:06:29.000000000 -0300
@@ -609,18 +609,18 @@
                 break;
 
             *NetNbItems += 1;
-            if( erc == OK )         // 1 marqueur par pin maxi
-            {
-                erc = DiagErc[ref_elect_type][jj];
-                if( erc != OK )
-                {
-                    if( g_NetObjectslist[NetItemTst]->m_FlagOfConnection == 0 )
-                    {
-                        Diagnose( panel, g_NetObjectslist[NetItemRef], g_NetObjectslist[NetItemTst], 0, erc );
-                        g_NetObjectslist[NetItemTst]->m_FlagOfConnection = NOCONNECT_SYMBOL_PRESENT;
-                    }
-                }
-            }
+            if( erc != OK )         // 1 marqueur par pin maxi
+                break;
+
+            erc = DiagErc[ref_elect_type][jj];
+            if( erc == OK )
+                break;
+
+            if( g_NetObjectslist[NetItemTst]->m_FlagOfConnection > 0 )
+                break;
+
+            Diagnose( panel, g_NetObjectslist[NetItemRef], g_NetObjectslist[NetItemTst], 0, erc );
+            g_NetObjectslist[NetItemTst]->m_FlagOfConnection = NOCONNECT_SYMBOL_PRESENT;
             break;
         }
     }
